packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -180,6 +222,209 @@ export const calculateFieldTransformer: DataTransformerInfo<CalculateFieldTransf
   },
 };
 
+function getWindowCreator(options: WindowOptions, allFrames: DataFrame[]): ValuesCreator {
+  if (options.windowSize! <= 0) {
+    throw new Error('Add field from calculation transformation - Window size must be larger than 0');
+  }
+
+  let matcher = getFieldMatcher({
+    id: FieldMatcherID.numeric,
+  });
+
+  if (options.field) {
+    matcher = getFieldMatcher({
+      id: FieldMatcherID.byNames,
+      options: {
+        names: [options.field],
+      },
+    });
+  }
+
+  const info = fieldReducers.get(options.reducer);
+
+  if (!info) {
```

Comment:
is this error being handled in the switch?

Reply:
These will be passed and caught in the panel error boundary. But when looking at this I decided to change this a bit, and improve the error message.